### PR TITLE
apps: replace legacy client with generated in pkg/deploy/cmd

### PR DESCRIPTION
**Pull request by mfojtik**
Ignore first 3 commits, they are part of https://github.com/openshift/origin/pull/16100

@deads2k last bits in apps group

### CODE REVIEW

File: pkg/deploy/cmd/delete.go [hunk 1]
```diff
@@ -9,30 +9,52 @@ import (
...
-	oc                    client.Interface
+	appsClient            appsclient.Interface
 	kc                    kclientset.Interface
 	pollInterval, timeout time.Duration
 }
 
+type updateConfigFunc func(d *deployapi.DeploymentConfig)
+
+// updateConfigWithRetries will try to update a deployment config and ignore any update conflicts.
+func updateConfigWithRetries(dn appsinternal.DeploymentConfigsGetter, namespace, name string, applyUpdate updateConfigFunc) (*deployapi.DeploymentConfig, error) {
```
Comment (mfojtik) [inline pkg/deploy/cmd/delete.go]: this should probably exists in some helper/util package

Comment (mfojtik) [reply pkg/deploy/cmd/delete.go]: @tnozicka do you know if we have this helperized already? i'm pretty sure we doing this in couple places.

File: pkg/deploy/cmd/history.go [hunk 2]
```diff
@@ -13,20 +13,18 @@ import (
...
-	return &DeploymentConfigHistoryViewer{dn: oc, rn: kc.Core()}
+func NewDeploymentConfigHistoryViewer(kc kclientset.Interface) kubectl.HistoryViewer {
+	return &DeploymentConfigHistoryViewer{rn: kc.Core()}
 }
 
 // DeploymentConfigHistoryViewer is an implementation of the kubectl HistoryViewer interface
 // for deployment configs.
 type DeploymentConfigHistoryViewer struct {
 	rn kcoreclient.ReplicationControllersGetter
-	dn client.DeploymentConfigsNamespacer
```
Comment (deads2k) [inline pkg/deploy/cmd/history.go]: this was just dead?  weird.

Comment (mfojtik) [reply pkg/deploy/cmd/history.go]: yeah... maybe a leftover from some refact.

File: pkg/deploy/cmd/delete.go [hunk 3]
```diff
@@ -9,30 +9,52 @@ import (
...
 
 // DeploymentConfigReaper implements the Reaper interface for deploymentConfigs
 type DeploymentConfigReaper struct {
-	oc                    client.Interface
+	appsClient            appsclient.Interface
 	kc                    kclientset.Interface
 	pollInterval, timeout time.Duration
 }
 
+type updateConfigFunc func(d *deployapi.DeploymentConfig)
```
Comment (deads2k) [inline pkg/deploy/cmd/delete.go]: Not a fan.  Why do we still  have this?

File: pkg/deploy/cmd/delete.go (hunk 1 above)
Comment (deads2k) [inline pkg/deploy/cmd/delete.go]: This just shouldn't exist.  If you want patch behavior, write a patch.

If this was pre-existing, open an issue.

File: pkg/deploy/cmd/delete.go (hunk 3 above)
Comment (mfojtik) [reply pkg/deploy/cmd/delete.go]: was just copy from the old client, seems like it is only used in one place, so right, we don't need this.

File: pkg/deploy/cmd/delete.go (hunk 1 above)
Comment (mfojtik) [reply pkg/deploy/cmd/delete.go]: lets fix that, low hanging fruit and I will forget about it: https://github.com/openshift/origin/pull/16166

Comment (tnozicka) [reply pkg/deploy/cmd/delete.go]: haven't seen such helper yet

File: pkg/cmd/util/clientcmd/factory_object_mapping.go [hunk 4]
```diff
@@ -221,11 +222,15 @@ func (f *ring1Factory) LogsForObject(object, options runtime.Object, timeout tim
 
 func (f *ring1Factory) Scaler(mapping *meta.RESTMapping) (kubectl.Scaler, error) {
 	if deployapi.IsKindOrLegacy("DeploymentConfig", mapping.GroupVersionKind.GroupKind()) {
-		oc, kc, err := f.clientAccessFactory.Clients()
+		_, kc, err := f.clientAccessFactory.Clients()
```
Comment (soltysh) [inline pkg/cmd/util/clientcmd/factory_object_mapping.go]: I don't like this. Are we planning to change `Clients()` to return openshift clientsets in the end? I do hope. Although probably it'll happen only when we actually remove `pkg/client`

**Comment by soltysh** [comment]
There are no actual api changes here, just comments for the generation. Applying the api-approved label. 
/lgtm

File: pkg/cmd/util/clientcmd/factory_object_mapping.go (hunk 4 above)
Comment (deads2k) [reply pkg/cmd/util/clientcmd/factory_object_mapping.go]: > I don't like this. Are we planning to change Clients() to return openshift clientsets in the end? I do hope. Although probably it'll happen only when we actually remove pkg/client

I think we will remove `Clients` altogether and create separate helpers for each individual type of client.

**Pull request merged**